### DISCUSSION TITLE
Integrate Swiss Coordinates and Conversion Tool in Locality Table

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#6003] [Feature Requests]
**Is your feature request related to a problem? Please describe.**  
There is a growing need among users involved in the SwissCollNet project for the inclusion of Swiss coordinates in the `Locality` table. Currently, Swiss users face challenges in managing and referencing Swiss coordinates effectively within the system since they do not use WGS84 by default.

**Describe the solution you'd like**  
Alexis proposes the addition of two dedicated fields in the locality table: **"Swiss Coordinate X"** and **"Swiss Coordinate Y."** 

There would be a conversion tool that automatically populates the corresponding WGS84 coordinates in their respective fields if they are empty. Furthermore, we request the inclusion of a button that links to the external website [[MAP GEOADMIN](https://map.geo.admin.ch/)](https://map.geo.admin.ch) to allow users to easily display the Swiss coordinates on the national reference map.

**Describe alternatives you've considered**  
While they have considered manually converting Swiss coordinates to WGS84 or using third-party tools, these methods are inefficient and prone to error. Integrating this functionality directly into our system would streamline the process and enhance user experience for Swiss users.

This is somewhat of an alternative approach to https://github.com/specify/specify7/issues/4761, so I would be in favor of first exploring this to have a direct integration as well as a link to the external site for more options.

**Reported By**  
Alexis Beck at Geneva

**Comment by beach53** [comment]
Copying this to Aimee who has had much experience with coordinate and
spatial data transformations.

I am a little hesitant about putting in a custom coordinate convertor for
one country into Specify's code.  There are many country systems, different
kinds of grids, projected and unprojected coordinate systems.  Would we be
doing a one-off with no extensibility?  I think this request from Geneva
needs broader discussion particularly in regard to how this fits in our
philosophy to be general and adaptable in implementing new capabilities.
Might be OK, but needs more discussion.

Jim

**Comment by grantfitzsimmons** [comment]
@beach53 This GitHub issue just represents a request submitted by Geneva that should be vetted, as I agree with your points.

<img width="1319" alt="image" src="https://github.com/user-attachments/assets/6aedc7df-db4f-4119-a983-c96f88b26624">

Without any code changes, you can use Web Links and some magic to add a couple of "swiss coordinate" fields and have it link directly to the https://map.geo.admin.ch/ ✨ 

<img width="838" alt="image" src="https://github.com/user-attachments/assets/6cec5e7e-d907-43fb-9611-31a230830cfb">

**Comment by grantfitzsimmons** [comment]
>There are many country systems, different
kinds of grids, projected and unprojected coordinate systems.  Would we be
doing a one-off with no extensibility?

We could make the application responsive to different datums and coordinate systems. I don't believe a solution only for a single region is the best option, of course.

**Comment by g1rly-c0d3r** [comment]
linked in https://github.com/specify/specify7/issues/8176
